### PR TITLE
Add support for Android with colors.android-colors.xml

### PR DESCRIPTION
**Pull request by kaelig**
@realdm I've added an XML file with a format that should work in Android. Can you test that this format would work for you?

### CODE REVIEW

File: dist/colors.android.xml [hunk 1]
```diff
@@ -0,0 +1,61 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<resources>
+  <property name="COLOR-PURPLE-LIGHTER" category="background-color" value="#fff6f0fd"/>
```
Comment (realdm) [inline dist/colors.android.xml]: Hi @kaelig please find a sample of how the resources are formatted for usage on android:
` <color name="colorPrimary">@color/polaris_titlebar_dark</color>`

Adding @see-mack  who is working on the polaris components for Android.

Comment (see-mack) [reply dist/colors.android.xml]: More specifically, a proper format with a hex value would be something like `<color name="polaris_indigo">#5c6ac4</color>`

This is the resource file we currently use: https://github.com/Shopify/android/blob/master/Polaris/Core/src/main/res/values/colors.xml

**Comment by kaelig** [comment]
I just updated the PR to follow the format used in the Android project.

It only contains the default Polaris palette. The way other projects deal with this is by extending the default palette with their project-specific colors. Is that something you could do as well?

File: dist/colors.android.xml (hunk 1 above)
Comment (kaelig) [reply dist/colors.android.xml]: @see-mack I updated the PR, can you please have another look?

**Comment by see-mack** [comment]
> It only contains the default Polaris palette. The way other projects deal with this is by extending the default palette with their project-specific colors. Is that something you could do as well?

Yes, we can add a 2nd color resources files to our project that references the colors defined in the file you've generated here.

**Comment by see-mack** [comment]
This looks great! If we can just change the indent to 4 spaces, we'll be good to go.

File: formats/android-colors.xml.js [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+              name: `polaris_${_.snakeCase(prop.get('name'))}`,
+            },
+          },
+          prop.get('value'),
+        ],
+      }))
+      .toJS(),
+  };
+  return xml(obj, {
+    indent: '  ',
```
[the remark this answers is not in the document]
Comment (kaelig) [reply formats/android-colors.xml.js]: For sure!

**Comment by kaelig** [comment]
Indentation is fixed. Your linter may warn about a missing line at the end of file.

Is there a way the linter could ignore the file? In a lot of projects, third-party dependencies don't get linted (for example, `node_modules` is ignored in Node apps).

**Comment by see-mack** [comment]
haha sorry to be a pain but the indent is 5 spaces :)

The linter can definitely be set to ignore certain files but I wouldn't consider this a 3rd party dependency since it's being generated by Shopify. AFAIK the line at the end of the file isn't a big deal but the 4 space indentation is considered important.

**Comment by kaelig** [comment]
> haha sorry to be a pain but the indent is 5 spaces :)

Sorry, that was embarrassing! Fixed :)

Any idea how you intend to pull this file into your project?

**Comment by see-mack** [comment]
> Any idea how you intend to pull this file into your project?

I haven't looked into it but we should be able to add a build step which grabs a specific version of the color token file and drops it into the project.

**Comment by kaelig** [comment]
Thank you! @DanielJette can help with the method to pull tokens in the Android repo.

**Comment by kaelig** [comment]
Thank you @see-mack and @realdm for the help reviewing this!

Will ship a new version of this package/gem soon :)

**Pull request merged**